src/site/topics/objects-review/equals.rst

```diff
@@ -1,56 +1,167 @@
+    ``someObject`` class' ``equals`` method.
+
+    ``Objects.equals(someObject, someOtherObject)`` also checks if ``someObject`` is ``null``, and then checks if
+    ``someObject`` and ``someOtherObject`` are equivalent based on ``someObject`` class' ``equals`` method.
+
+    The below example makes use of the third option to be safe around ``null``, but realistically, based on the way the
+    ``Friend`` class is written, it is not possible for an instance of a ``Friend`` object to be created with ``null``
+    fields. In other words, the second option would be fine, but using the third option would still be more defensive.
+
+
+.. code-block:: java
+    :linenos:
+
+    /**
+     * Checks if two Friend objects are equal. Friend objects are considered equal if all their attributes are equal.
+     *
+     * @param o an "object" being compared to.
+     * @return True if the two objects are equal, false otherwise.
+     */
+    public boolean equals(Object o) {
+        // If o is actually in the same memory address of this
+        if (o == this) {
+            return true;
+        }
+        // If o is null, then it's not equal
+        if (o == null) {
+            return false;
+        }
+        // if o and this are of different classes, they're not the equal
+        if (o.getClass() != this.getClass()) {
+            return false;
+        }
+        // Cast o as a friend
+        Friend other = (Friend) o;
+        return Objects.equals(this.firstName, other.firstName) &&
+                Objects.equals(this.lastName, other.lastName) &&
+                Objects.equals(this.email, other.email);
+    }
+
+
+* Below is an example of the ``equals`` method for the ``ContactList`` class
+
+    * Notice that it follows the same pattern
+    * The only difference is the check on what needs to be equal
+    * Here, ``Arrays.equals`` is used to check the equality on the array
+
+        * The alternative would be to loop over the array and check equality on each element within the loop
+
+
+.. code-block:: java
+    :linenos:
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ContactList that = (ContactList) o;
+        return this.size == that.size && Arrays.equals(this.friends, 0, this.size(), that.friends, 0, that.size());
+    }
+
+
+
+Hash Code
+=========
+
+* When properly writing the ``equals`` method, one should also write another special method --- ``hashCode()``
+
+    * The full details on what ``hashCode`` is and what it is for is beyond the scope of this course
+    * Briefly, it is a function used to convert the object into an ``int`` hash value
+    * Any two objects that are equal must have the same hash value
+    * Ideally, the hash value will be unique --- any *unequal* objects should have different hash values
```

Comment:
More like _should aim to have different hashes_. Clearly the set of strings is larger than the set of `int`s, so collisions are inevitable.